src/linear_algebra/char_poly/coeff.lean

```diff
@@ -203,3 +203,16 @@ by rw [trace_eq_neg_char_poly_coeff, trace_eq_neg_char_poly_coeff,
 lemma zmod.trace_pow_card {p:ℕ} [fact p.prime] [nonempty n] (M : matrix n n (zmod p)) :
   trace n (zmod p) (zmod p) (M ^ p) = (trace n (zmod p) (zmod p) M)^p :=
 by { have h := finite_field.trace_pow_card M, rwa zmod.card at h, }
+
+namespace matrix
+
+theorem is_integral : is_integral R M := ⟨char_poly M, ⟨char_poly_monic M, aeval_self_char_poly M⟩⟩
+
+/-- The minimal polynomial of a matrix -/
+def min_poly : polynomial R := minimal_polynomial (M.is_integral)
```

Comment:
I'm not so sure how useful this definition will be: if people want to use the properties of `minimal_polynomial` to reason about `matrix.min_poly`, they will have to unfold `matrix.min_poly` first. I expect it will be less work to use `minimal_polynomial M.is_integral` everywhere. (And because `is_integral R M` is a `Prop`, there should be no risk of confusion.)